mlrun/package/context_handler.py

```diff
@@ -323,3 +305,52 @@ def _collect_mlrun_packagers(self):
                 is_mandatory=False,
                 is_custom_packagers=False,
             )
+
+    def _is_logging_worker(self):
+        """
+        Check if the current worker is the logging worker.
+
+        :return: True if the context belongs to the logging worker and False otherwise.
+        """
+        # If it's a OpenMPI job:
+        if self._context.labels.get("kind", "job") == "mpijob":
```

Comment:
Checking for the job kind is not really good practice because we want the ContextHandler to be agnostic to the run kind. Maybe we should move this to the runtime handlers (MpiV1RuntimeHandler and MpiV1Alpha1RuntimeHandler) and use get_runtime_handler to get the handler

Reply:
I need you to explain the role of this RuntimeHandler first then, because I've never seen it :)

Reply:
RuntimeHandlers moved to the API so comment is no longer relevant.
We can move this to the Runtime class (If in the future we'll have this logic for more runtimes) but for now I'm ok with this